lib/Language/Souffle/Interpreted.hs

```diff
@@ -36,14 +38,22 @@ import System.Process
 import Text.Printf
 
 
-newtype SouffleM a = SouffleM { runSouffle :: IO a }
+runSouffle :: SouffleM a -> IO a
+runSouffle (SouffleM m) = runReaderT m Nothing
+
+-- | Run a souffle interpreter that will look up the datalog program
+-- in the given directory.
+runSouffleWith :: FilePath -> SouffleM a -> IO a
+runSouffleWith datalogProgramPath (SouffleM m) = runReaderT m (Just datalogProgramPath)
+
+newtype SouffleM a = SouffleM (ReaderT (Maybe FilePath) IO a)
```

Comment:
I'd get rid of the Maybe here and call `runSouffleWith` from `runSouffle`. We can consider a config object as well (contains souffleBin, datalogProgramPath, ...) and a default config object for ease of use.